From: Hans de Goede <hdegoede@redhat.com>
Date: Thu, 11 Feb 2016 13:11:48 +0100
Subject: get_media_devices.c: Remove superfluous ; add end of functions

Signed-off-by: Hans de Goede <hdegoede@redhat.com>
---
 common/get_media_devices.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/common/get_media_devices.c b/common/get_media_devices.c
index a2d6354..619734e 100644
--- a/common/get_media_devices.c
+++ b/common/get_media_devices.c
@@ -227,7 +227,7 @@ static int add_v4l_class(struct media_device_entry *md)
 		md->type = MEDIA_V4L_SUBDEV;
 
 	return 0;
-};
+}
 
 static int add_snd_class(struct media_device_entry *md)
 {
@@ -270,7 +270,7 @@ static int add_snd_class(struct media_device_entry *md)
 	md->node = strdup(node);
 
 	return 0;
-};
+}
 
 static int add_dvb_class(struct media_device_entry *md)
 {
@@ -294,7 +294,7 @@ static int add_dvb_class(struct media_device_entry *md)
 		md->type = MEDIA_DVB_OSD;
 
 	return 0;
-};
+}
 
 static int sort_media_device_entry(const void *a, const void *b)
 {
